Added the docs for all the grafana dashboards. (#21795)
hashicorp/consul: website/content/docs/connect/observability/grafanadashboards/consulk8sdashboard.mdx

---
layout: docs
page_title: Dashboard for Consul k8s control plane metrics
description: >-
  This documentation provides an overview of the Consul on Kubernetes Dashboard
---

# Consul on Kubernetes control plane monitoring dashboard

> This page provides reference information about the Grafana dashboard configuration included in [this GitHub repository](https://github.com/hashicorp/consul/tree/main/grafana/consulk8sdashboard.json).

Reviewer: The github link is wrong

---
layout: docs
page_title: Dashboard for Consul k8s control plane metrics
description: >-
  This documentation provides an overview of the Consul K8s Dashboard
---

# Consul k8s monitoring (Control Plane) dashboard

### Number of Consul servers

> - **Grafana query:** `consul_consul_server_0_consul_members_servers{pod="consul-server-0"}`
> - **Description:** Displays the number of Consul servers currently active. This metric provides insight into the cluster's health and the number of Consul nodes running in the environment.

Reviewer 2: ```suggestion
- **Grafana query:** `consul_consul_server_0_consul_members_servers{pod="consul-server-0"}`
- **Description:** Displays the number of Consul servers currently active. This metric provides insight into the cluster's health and the number of Consul nodes running in the environment.
```
This only retrieves the metric from the pod named `consul-server-0`. Is it possible to modify this so that the value is retrieved from any available server, or the Raft leader?

### Number of connected Consul dataplanes

- **Grafana query:** `count(consul_dataplane_envoy_connected)`
- **Description:** Tracks the number of connected Consul dataplanes. This metric helps operators understand how many Envoy sidecars are actively connected to the mesh.

### CPU usage in seconds (Consul servers)

- **Grafana query:** `rate(container_cpu_usage_seconds_total{container="consul", pod=~"consul-server-.*"}[5m])`

> - **Description:** This metric shows the CPU usage of the Consul servers over time, helping operators monitor resource consumption.

Reviewer 2: ```suggestion
- **Description:** This metric shows the CPU usage of the Consul servers over time, helping monitor resource consumption.
```

### Memory usage (Consul servers)

- **Grafana query:** `container_memory_working_set_bytes{container="consul", pod=~"consul-server-.*"}`
- **Description:** Displays the memory usage of the Consul servers. This metric helps ensure that the servers have sufficient memory resources for proper operation.

### Disk read/write total per 5 minutes (Consul servers)

- **Grafana query:** `sum(rate(container_fs_writes_bytes_total{pod=~"consul-server-.*",
 container="consul"}[5m])) by (pod, device)`
- **Grafana query:** `sum(rate(container_fs_reads_bytes_total{pod=~"consul-server-.*", container="consul"}[5m])) by (pod, device)`

> - **Description:** Monitors disk read and write operations over 5-minute intervals for Consul servers. This helps identify potential disk bottlenecks or issues.

Reviewer 2: ```suggestion
- **Description:** Monitors disk read and write operations over 5-minute intervals for Consul servers. Use this metric to identify potential disk I/O bottlenecks or throughput issues.
```

### Memory limit (Consul servers)

- **Grafana query:** `kube_pod_container_resource_limits{resource="memory", pod="consul-server-0"}`
- **Description:** Displays the memory limit for Consul servers. This metric ensures that memory usage stays within the defined limits for each Consul server.

### CPU limit in seconds (Consul servers)

- **Grafana query:** `kube_pod_container_resource_limits{resource="cpu", pod="consul-server-0"}`
- **Description:** Displays the CPU limit for Consul servers. Monitoring CPU limits helps operators ensure that the services are not constrained by resource limitations.

### Disk usage (Consul servers)

- **Grafana query:** `sum(container_fs_usage_bytes{}) by (pod)`
- **Grafana query:** `sum(container_fs_usage_bytes{pod="consul-server-0"})`
- **Description:** Shows the amount of filesystem storage used by Consul servers. This metric helps operators track disk usage and plan for capacity.

### CPU usage in seconds (Connect injector)

- **Grafana query:** `rate(container_cpu_usage_seconds_total{pod=~".*-connect-injector-.*",
container="sidecar-injector"}[5m])`

> - **Description:** Tracks the CPU usage of the Connect Injector, which is responsible for injecting Envoy sidecars. Monitoring this helps ensure that Connect Injector has adequate CPU resources.

Reviewer 2: ```suggestion
- **Description:** Tracks the CPU usage of the Connect Injector, which is responsible for injecting Envoy sidecars and other operations within the mesh. Monitoring this helps ensure that Connect Injector has adequate CPU resources.
```
The `connect-injector` process also acts as the controller for API Gateway.

layout: docs
page_title: Dashboard for Consul k8s control plane metrics
description: >-

>   This documentation provides an overview of the Consul Kubernetes Dashboard

Reviewer 3: nit: should this read `Consul on Kubernetes`?
